nomt/src/beatree/branch/node.rs

```diff
@@ -77,30 +80,31 @@ impl BranchNode {
         slice[6..8].copy_from_slice(&len.to_le_bytes());
     }
 
-    pub fn separator_len(&self) -> u16 {
-        self.view().separator_len()
+    pub fn prefix(&self) -> &BitSlice<u8, Msb0> {
+        self.view().prefix()
     }
 
-    pub fn set_separator_len(&mut self, len: u16) {
-        let slice = self.as_mut_slice();
-        slice[8..10].copy_from_slice(&len.to_le_bytes());
+    fn set_prefix(&mut self, prefix: &BitSlice<u8, Msb0>) {
+        let start = BRANCH_NODE_HEADER_SIZE + (self.n() as usize + 1) * 2;
+        let prefix_len = self.prefix_len() as usize;
+        self.as_mut_slice()[start..].view_bits_mut()[..prefix_len].copy_from_bitslice(prefix);
     }
 
-    fn varbits_mut(&mut self) -> &mut BitSlice<u8, Msb0> {
-        let body_end = body_size(
-            self.prefix_len() as _,
-            self.separator_len() as _,
-            self.n() as _,
-        ) + BRANCH_NODE_HEADER_SIZE;
-        self.as_mut_slice()[BRANCH_NODE_HEADER_SIZE..body_end].view_bits_mut()
+    pub fn separator(&self, i: usize) -> &BitSlice<u8, Msb0> {
+        self.view().separator(i)
     }
 
-    pub fn prefix(&self) -> &BitSlice<u8, Msb0> {
-        self.view().prefix()
-    }
+    fn set_separators(&mut self, cells: Vec<u8>, separators: BitVec<u8, Msb0>) {
```

Comment:
this also seems unnecessary if you can get rid of the allocations in `BranchBuilder`.